synapse/storage/databases/main/stats.py

```diff
@@ -339,7 +347,7 @@ async def update_stats_delta(
         stats_type: str,
         stats_id: str,
         fields: Dict[str, int],
-        complete_with_stream_id: Optional[int],
+        complete_with_stream_id: int,
```

Comment:
This `Optional[int]` is in conflict with
https://github.com/matrix-org/synapse/blob/5e8127c22f8c0f0b7d3ad3d23079761163975871/synapse/storage/databases/main/stats.py#L419-L421

I could not see where `Optional` is needed.

Reply:
`update_stats_delta` has two callers, one calculates `complete_with_stream_id` by calling `get_room_max_stream_ordering()`, the other from `_get_max_stream_id_in_current_state_deltas_txn` which will always return an `int`, so I agree. 👍